gtsfm/scene_optimizer.py

```diff
@@ -96,8 +133,30 @@ def create_computation_graph(
         camera_intrinsics_i1_graph: Delayed,
         camera_intrinsics_i2_graph: Delayed,
         exact_intrinsics: bool = True,
-    ) -> Tuple[Delayed, Delayed, Delayed]:
-        """Create delayed tasks for matching and verification."""
+        i2Ti1_expected_graph: Optional[Delayed] = None,
+    ) -> Tuple[Delayed, Delayed, Delayed, Optional[Delayed], Optional[Delayed]]:
+        """Create delayed tasks for matching and verification.
+
+        Args:
+            keypoints_i1_graph: keypoints for image i1.
+            keypoints_i2_graph: keypoints for image i2.
+            descriptors_i1_graph: corr. descriptors for image i1.
+            descriptors_i2_graph: corr. descriptors for image i2.
+            camera_intrinsics_i1_graph: intrinsics for camera i1.
+            camera_intrinsics_i2_graph: intrinsics for camera i2.
+            exact_intrinsics (optional): flag to use intrinsics as exact.
+                                         Defaults to True.
+            i2Ti1_expected_graph (optional): ground truth relative pose, used
+                                             for evaluation if available.
+                                             Defaults to None.
+
+        Returns:
+            computed relative rotation wrapped as Delayed.
+            computed relative translation direction wrapped as Delayed.
+            indices of verified correspondences wrapped as Delayed.
+            error in relative rotation wrapped as Delayed
+            error in relative translation direction wrapped as Delayed.
```

Comment:
It might be better to have a different accessor (or property in the python world) for the errors. We should try to and keep the API for TwoViewEstimator as close to its functionality as possible. Now someone who does not want the errors will need dummy variables in their code.

Reply:
Nice point, but I'll need your suggestions on how to organize it.

We need the logging to be in this file, so that leaves us with just 2 lines of code (calling the two error functions in `comp_utils`). I can remove this function, and copy those two lines to the `create_computation_graph` function?

I thought having a separate function just allows cleaner and smaller functions, even if they call each other.

Reply:
That is not what I was talking about in this comment. It's good to have the _compute_metrics method. This comment is about not returning the errors in `create_computation_graph`. You should still compute them here, but store them as member variables and return them in another get_metrics() method.

Reply:
I have to return it in the `create_computation_graph` function because it has to be made part of the main computation in the scene_optimizer class. The user anyway doesn't have to interact/change these files at all so I personally feel having a single function is the best option

Reply:
Not sure if we can store side effects in member variables with Dask since everything seems functional in the computational graph